cmd/ocitools/validate.go

```diff
@@ -397,6 +398,26 @@ func checkLinuxResources(r rspec.Resources, hostCheck bool) (msgs []string) {
 			msgs = append(msgs, fmt.Sprintf("Minimum memory limit should be larger than memory reservation"))
 		}
 	}
+	if r.Network != nil && hostCheck {
+		var exist bool
+		interfaces, err := net.Interfaces()
+		if err != nil {
+			msgs = append(msgs, err.Error())
+			return
+		}
+		for _, prio := range r.Network.Priorities {
+			exist = false
+			for _, ni := range interfaces {
+				if prio.Name == ni.Name {
+					exist = true
+					break
+				}
+			}
+			if !exist {
+				msgs = append(msgs, fmt.Sprintf("Interface %s does not exist on host", prio.Name))
```

Comment:
On Tue, Aug 23, 2016 at 06:42:57PM -0700, Ma Shimiao wrote:
> > +		for _, prio := range r.Network.Priorities {
> > +			exist = false
> > +			for _, ni := range interfaces {
> > +				if prio.Name == ni.Name {
> > +					exist = true
> > +					break
> > +				}
> > +			}
> > +			if !exist {
> > +				msgs = append(msgs, fmt.Sprintf("Interface %s does not exist on host", prio.Name))
> 
> It seems hooks can't populate a new network namespace.

Why not?  I think this is one of the main use-cases for hooks or other
post-create, pre-start activity.  The problem here is that these
cgroup settings are applied during create before those hooks or
post-create actions have fired.

> And runtime SPEC doesn't say how container can setup new network
> interface in a new network namespace.

Agreed.

> So, I think we can only set priorities for host interfaces.

Or if you join an existing interface, since with either of those the
interfaces will exist when the cgroup priorities, etc. are setup.

I'm not clear on how configuring a new cgroup limiting interfaces from
a joined namespace interacts with the language from
opencontainers/runtime-spec#158.  It certainly seems to fall under a
literal interpretation of “anything else related to that namespace”.
But the spirit of that limitation (as I see it) was to avoid
contention for shared resources (e.g. multiple containers trying to
set the hostname in a shared UTS namespace), and if you're creating
and configuring a new cgroup you aren't stepping on another
container's toes.